src/coreclr/nativeaot/System.Private.CoreLib/src/System/Array.CoreRT.cs

```diff
@@ -581,7 +581,7 @@ private static unsafe void CopyImplValueTypeArrayWithInnerGcRefs(Array sourceArr
 
                     object boxedValue = RuntimeImports.RhBox(sourceElementEEType, ref *pSourceElement);
                     if (reliable)
-                        boxedElements[i] = boxedValue;
+                        boxedElements![i] = boxedValue;
```

Comment:
We can avoid `!` here by changing the condition to `if (boxedElements != null)`

Reply:
Great!